Helper/Data.php

```diff
@@ -95,13 +95,12 @@ protected function arrayFlatten(array $array, $parentKey = '')
     {
         $result = [];
         foreach ($array as $key => $value) {
-            if (is_object($value)) {
+            if (is_array($value)) {
                 foreach ($value as $k => $v) {
-                    $result[] = $key;
-                    $result[] = $k . $v;
+                    $result[$key . $k] = $v;
                 }
             } else {
-                $result[] = $key . $value;
+                $result[$key] = $value;
```

Comment:
The logic changed a bit from what we talked about. Are you sure we want to set `$key` as the key? It changes this from an array to an associative array if i'm not mistaken.

Reply:
Yes, you are right, I did this to minimize the changes because in this extension, right after calling this function, "join" is used to concat the keys with the values.

The full example of what's happening:

```
<?php
        //Enter your code here, enjoy!

$payload = json_decode('{ "event_type": "ORDER.PAYMENT.RECEIVED", "resource": { "amount": "8.64", "currency": "USD", "reference": "6000142632" }, "state": "completed" }', true);

function arrayFlatten(array $array, $parentKey = '')
    {
        $result = [];
        foreach ($array as $key => $value) {
            if (is_array($value)) {
                foreach ($value as $k => $v) {
                    $result[$key . $k] = $v;
                }
            } else {
                $result[$key] = $value;
            }
        }
        return $result;
    }

function _joinArrayKeysAndValues(array $array)
{
    $result = "";
    $result = join(
        "",
        array_map(
            function ($v, $k) {
                return $k . $v;
            },
            $array,
            array_keys($array)
        )
    );

    return $result;
}

// Concat nested keys to a flatten array
$payloadConcatedKeys = arrayFlatten($payload);

// Sort array alphabetically
ksort($payloadConcatedKeys);

// Concat the array
$concatedPayload = _joinArrayKeysAndValues($payloadConcatedKeys);

var_dump($concatedPayload);
```

Reply:
Oh, ok! That makes more sense.